### DISCUSSION TITLE
Pull requests reviews for user

### DISCUSSION THREAD

**Discussion by ilia-cy** [community/community#24546] [Apps, API and Webhooks]
<p>I’m trying to get all pull request reviews for a specific user (If needed, I can supply a specific repository on which to search).</p>
<p>I tried using the Rest API, but all you can get there is:</p>
<ol>
<li>
<p>Pull requests for a given repository and from there you can get the reviews (if exists). No filter per user.</p>
</li>
<li>
<p>With a pull request id, get its reviewes. No filter per user.</p>
</li>
</ol>
<p>Using the GraphQL API, I was able to reach the following query:</p>
<pre><code>query {
    user(login: &lt;userName&gt;) {
        contributionsCollection {
            pullRequestReviewContributionsByRepository {
                contributions(first: 100) {
                    nodes {
                        pullRequestReview {
                            &lt;DATA HERE&gt;
                        }
                    }
                }
            }
        }
    }
}
</code></pre>
<p>But as it seems, it only returns data for public repositories (or my own pull request reviews), even when I’m using an authenticated token for an organization.</p>
<p>Using that token and supplying some member from the oranization, the <code>pullRequestReviewContributionsByRepository</code> parameter always returns as an empty array.</p>
<p>Is there any easy way of getting all pull request reviews for a given user (on a given repository will also work)?</p>
<p>Thanks</p>

**Comment by ilia-cy** [comment]
<p>Up until now, the only approach I found is the following:</p>
<pre><code>query {
  repository(name: &lt;repoName&gt;, owner: &lt;orgName&gt;) {
    pullRequests(first: 100) {
      nodes {
        reviews(first: 100, author: &lt;userName&gt;) {
          nodes {
            &lt;DATA HERE&gt;
          }
        }
      }
    }
  }
}
</code></pre>
<p>This will provide the pull requests reviewed for a given user on a given repository, but in order to get to these you will have to filter out all the other pull requests in the repository (that did not hold a review by the author).</p>
<p>See response for example:</p>
<pre><code>{
  "data": {
    "repository": {
      "pullRequests": {
        "nodes": [
          {
            "reviews": {
              "nodes": []
            }
          },
          {
            "reviews": {
              "nodes": []
            }
          },
          {
            "reviews": {
              "nodes": [
                {
                  "submittedAt": "2020-04-04T20:33:04Z"
                }
              ]
            }
          },
          {
            "reviews": {
              "nodes": [
                {
                  "submittedAt": "2020-04-04T18:19:22Z"
                }
              ]
            }
          },
          {
            "reviews": {
              "nodes": []
            }
          },
          {
            "reviews": {
              "nodes": []
            }
          },
          {
            "reviews": {
              "nodes": []
            }
          },
          {
            "reviews": {
              "nodes": []
            }
          },
          {
            "reviews": {
              "nodes": []
            }
          },
          {
            "reviews": {
              "nodes": []
            }
          }
        ]
      }
    }
  }
}
</code></pre>
<p>So if anyone has a less awkward approach, it would be appreciated.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/ilia-cy">@ilia-cy</a>,</p>
<p>Thanks for being here! Would <a href="https://developer.github.com/v4/object/contributionscollection/" rel="nofollow noopener"><code>ContributionsConnection</code></a> help?</p>
<pre><code>{
  user(login: "andreagriffiths11") {
    login
    contributionsCollection {
      commitContributionsByRepository {
        contributions(last: 100) {
          nodes {
            occurredAt
            commitCount
            repository {
              nameWithOwner
            }
          }
        }
      }
    }
  }
}
</code></pre>

**Comment by ilia-cy** [comment]
<p>Hello Andrea,</p>
<p>The contributionsCollection is indeed the right solution, as it provides all pull request reviews per repository.</p>
<p>Thing is, it only works for public content.</p>
<p>I discussed this with Github support, and they say it’s a known (non documented) issue and they currently don’t have any ETA of when it will be supported.</p>
